AssertionError: Cannot change span of EmptyTree when declaring abstract inline method

## Minimized code
https://scastie.scala-lang.org/8IwskpLVRfCD9OwzDpSG0g
```scala
trait Extractor {
  inline def unapplySeq(inline tn: String): Option[Seq[String]]
}

transparent inline def (inline sc: StringContext).poql: Extractor = new Extractor {
  inline def unapplySeq(inline tn: String): Option[Seq[String]] = ???
}

object x {
  "x" match {
    case poql" $x" => x
  }
}
```

## Output

```
java.lang.AssertionError: Cannot change span of EmptyTree
	at dotty.tools.dotc.ast.Trees$EmptyTree.withSpan(Trees.scala:848)
	at dotty.tools.dotc.ast.Trees$EmptyTree.withSpan(Trees.scala:848)
	at dotty.tools.dotc.typer.Inliner$$anon$11.transform(Inliner.scala:82)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1318)
	at dotty.tools.dotc.typer.Inliner$$anon$11.transform(Inliner.scala:82)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transformSub(Trees.scala:1346)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1325)
	at dotty.tools.dotc.typer.Inliner$$anon$11.transform(Inliner.scala:82)
	at dotty.tools.dotc.ast.Trees$Instance$TreeMap.transform(Trees.scala:1322)
	at dotty.tools.dotc.typer.Inliner$$anon$11.transform(Inliner.scala:82)
	at dotty.tools.dotc.typer.Inliner$.liftBindings$2$$anonfun$1(Inliner.scala:99)
	at scala.collection.immutable.List.map(List.scala:246)
	at dotty.tools.dotc.typer.Inliner$.liftBindings$1(Inliner.scala:94)
	at dotty.tools.dotc.typer.Inliner$.liftBindings$1(Inliner.scala:99)
	at dotty.tools.dotc.typer.Inliner$.liftBindings$1(Inliner.scala:105)
	at dotty.tools.dotc.typer.Inliner$.liftBindings$1(Inliner.scala:101)
	at dotty.tools.dotc.typer.Inliner$.inlineCall(Inliner.scala:110)
	at dotty.tools.dotc.typer.Inliner$.$anonfun$3(Inliner.scala:161)
	at dotty.tools.dotc.ast.tpd$.polyDefDef(tpd.scala:287)
	at dotty.tools.dotc.ast.tpd$.DefDef(tpd.scala:225)
	at dotty.tools.dotc.typer.Inliner$.inlinedUnapply(Inliner.scala:162)
	at dotty.tools.dotc.typer.Typer$$anon$1.transform(Typer.scala:1352)
	at dotty.tools.dotc.typer.Typer.caseRest$1(Typer.scala:1379)
	at dotty.tools.dotc.typer.Typer.typedCase(Typer.scala:1390)
	at dotty.tools.dotc.typer.Typer.typedCases$$anonfun$1(Typer.scala:1327)
	at dotty.tools.dotc.core.Decorators$ListDecorator$.loop$1(Decorators.scala:63)
	at dotty.tools.dotc.core.Decorators$ListDecorator$.mapconserve$extension(Decorators.scala:79)
	at dotty.tools.dotc.typer.Typer.typedCases(Typer.scala:1329)
	at dotty.tools.dotc.typer.Typer.$anonfun$18(Typer.scala:1319)
	at dotty.tools.dotc.typer.Applications.harmonic(Applications.scala:1996)
	at dotty.tools.dotc.typer.Typer.harmonic(Typer.scala:85)
	at dotty.tools.dotc.typer.Typer.typedMatchFinish(Typer.scala:1319)
	at dotty.tools.dotc.typer.Typer.typedMatch(Typer.scala:1284)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2358)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2404)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2443)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2452)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2455)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2500)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2522)
	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2003)
	at dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$2(Typer.scala:2333)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2337)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2403)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2443)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2452)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2455)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2477)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2522)
	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2129)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2376)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2404)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2443)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2452)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2455)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2566)
	at dotty.tools.dotc.typer.FrontEnd.liftedTree1$1(FrontEnd.scala:79)
	at dotty.tools.dotc.typer.FrontEnd.typeCheck$$anonfun$1(FrontEnd.scala:84)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:42)
	at dotty.tools.dotc.typer.FrontEnd.typeCheck(FrontEnd.scala:85)
	at dotty.tools.dotc.typer.FrontEnd.runOn$$anonfun$3(FrontEnd.scala:114)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.immutable.List.foreach(List.scala:333)
	at dotty.tools.dotc.typer.FrontEnd.runOn(FrontEnd.scala:114)
	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:165)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at dotty.tools.dotc.Run.runPhases$5(Run.scala:175)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:183)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:64)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:190)
	at dotty.tools.dotc.Run.compileSources(Run.scala:127)
	at dotty.tools.dotc.Run.compile(Run.scala:110)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:38)
	at dotty.tools.dotc.Driver.process(Driver.scala:194)
	at dotty.tools.dotc.Main.process(Main.scala)
	at xsbt.CachedCompilerImpl.run(CachedCompilerImpl.java:69)
	at xsbt.CompilerInterface.run(CompilerInterface.java:41)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:248)
	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:122)
	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:95)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:186)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3(MixedAnalyzingCompiler.scala:82)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3$adapted(MixedAnalyzingCompiler.scala:77)
	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:215)
	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:77)
	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:146)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:343)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:343)
	at sbt.internal.inc.Incremental$.doCompile(Incremental.scala:120)
	at sbt.internal.inc.Incremental$.$anonfun$compile$4(Incremental.scala:100)
	at sbt.internal.inc.IncrementalCommon.recompileClasses(IncrementalCommon.scala:180)
	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:98)
	at sbt.internal.inc.Incremental$.$anonfun$compile$3(Incremental.scala:102)
	at sbt.internal.inc.Incremental$.manageClassfiles(Incremental.scala:155)
	at sbt.internal.inc.Incremental$.compile(Incremental.scala:92)
	at sbt.internal.inc.IncrementalCompile$.apply(Compile.scala:75)
	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:348)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:301)
	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:168)
	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:248)
	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:74)
	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:1765)
	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:1738)
	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
	at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
	at sbt.Execute.work(Execute.scala:290)
	at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

## Expectation

Not sure whether it will compile, but didn't expect an AssertionError

--- a/tests/neg/i8894.scala
+++ b/tests/neg/i8894.scala
@@ -1,0 +1,13 @@
+trait Extractor {
+  inline def unapplySeq(inline tn: String): Option[Seq[String]]
+}
+
+transparent inline def (inline sc: StringContext).poql: Extractor = new Extractor {
+  inline def unapplySeq(inline tn: String): Option[Seq[String]] = ??? // error: Implementation restriction: nested inline methods are not supported
+}
+
+object x {
+  "x" match {
+    case poql" $x" => x // error: Deferred inline method unapplySeq in trait Extractor cannot be invoked
+  }
+}

